libraries/Netdump/src/NetdumpPacket.cpp

```diff
@@ -0,0 +1,251 @@
+        sstr.printf("\r\n");
+        printDetail(sstr, "           H ", &data[ETH_HDR_LEN + getIpHdrLen()], getUdpHdrLen(), netdumpDetail);
+        printDetail(sstr, "           D ", &data[ETH_HDR_LEN + getIpHdrLen() + getUdpHdrLen()], getUdpLen(), netdumpDetail);
+        return sstr;
+    }
+    case PacketType::TCP :
+    case PacketType::HTTP :
+    {
+        sstr.printf("%s>%s ", sourceIP().toString().c_str(), destIP().toString().c_str());
+        sstr.printf("%d:%d ", getSrcPort(), getDstPort());
+        uint16_t flags = getTcpFlags();
+        sstr.print('[');
+        const char chars [] = "FSRPAUECN";
+        for (uint8_t i = 0; i < sizeof chars; i++)
+            if (flags & (1 << i))
+            {
+                sstr.print(chars[i]);
+            }
+        sstr.print(']');
+        sstr.printf(" len: %u seq: %u, ack: %u, wnd: %u ", getTcpLen(), getTcpSeq(), getTcpAck(), getTcpWindow());
+        sstr.printf("\r\n");
+        printDetail(sstr, "           H ", &data[ETH_HDR_LEN + getIpHdrLen()], getTcpHdrLen(), netdumpDetail);
+        printDetail(sstr, "           D ", &data[ETH_HDR_LEN + getIpHdrLen() + getTcpHdrLen()], getTcpLen(), netdumpDetail);
+        return sstr;
+    }
+    case PacketType::ICMP :
+    {
+        sstr.printf("%s>%s ", sourceIP().toString().c_str(), destIP().toString().c_str());
+        if (isIPv4())
+        {
+            switch (getIcmpType())
+            {
+            case 0 : sstr.printf("ping reply"); break;
+            case 8 : sstr.printf("ping request"); break;
+            default: sstr.printf("type(0x%02x)", getIcmpType()); break;
+            }
+        }
+        if (isIPv6())
+        {
+            switch (getIcmpType())
+            {
+            case 129 : sstr.printf("ping reply"); break;
+            case 128 : sstr.printf("ping request"); break;
+            case 135 : sstr.printf("Neighbour solicitation"); break;
+            case 136 : sstr.printf("Neighbour advertisement"); break;
+            default: sstr.printf("type(0x%02x)", getIcmpType()); break;
+            }
+        }
+        sstr.printf("\r\n");
+        return sstr;
+    }
+    case PacketType::IGMP :
+    {
+    	switch (getIgmpType())
+        {
+        case 1 : sstr.printf("Create Group Request"); break;
+        case 2 : sstr.printf("Create Group Reply"); break;
+        case 3 : sstr.printf("Join Group Request"); break;
+        case 4 : sstr.printf("Join Group Reply"); break;
+        case 5 : sstr.printf("Leave Group Request"); break;
+        case 6 : sstr.printf("Leave Group Reply"); break;
+        case 7 : sstr.printf("Confirm Group Request"); break;
+        case 8 : sstr.printf("Confirm Group Reply"); break;
+        case 0x11 : sstr.printf("Group Membership Query"); break;
+        case 0x12 : sstr.printf("IGMPv1 Membership Report"); break;
+        case 0x22 : sstr.printf("IGMPv3 Membership Report"); break;
+        default: sstr.printf("type(0x%02x)", getIgmpType()); break;
```

Comment:
This may be one of the very few cases where it's ok to have single-line cases (table-like).
The nested switch should still be moved out to a function.